### PR TITLE
use globalThis instead of window in HMR

### PR DESCRIPTION
**Pull request by MaxArtemov**
Usage of `window` in non-browser env, breaks 

## Summary by CodeRabbit

- **新特性**
	- 改进了热模块替换（HMR）逻辑，增强了对不同环境的兼容性。
- **修复**
	- 添加了对`globalThis`的优先支持，以提高在非生产环境中的稳定性。

### CODE REVIEW

**Comment by antoniopresto** [comment]
`+ 1`

This  may fix `window is not defined` in next.js "edge runtimes", right?

for more context: https://wintercg.org/

> The ultimate goal of the group is to promote runtimes supporting a comprehensive unified API surface that JavaScript developers can rely on regardless of the runtime they are using: be it browsers, servers, embedded applications, or edge runtimes. The members of the group want to provide a space to better coordinate between browser vendors and other implementors on how Web Platform APIs can be best implemented and used outside of browsers.

File: src/hooks/useHMR.ts [hunk 1]
```diff
@@ -18,8 +18,8 @@ if (
   module &&
   module.hot
 ) {
-  const win = window as any;
-  if (typeof win.webpackHotUpdate === 'function') {
+  const win = globalThis as any;
+  if (typeof globalThis.webpackHotUpdate === 'function') {
```
Comment (antoniopresto) [inline src/hooks/useHMR.ts:22]:
```suggestion
  const win = globalThis as any;
  if (typeof win.webpackHotUpdate === 'function') {
```

**Comment by XavierLeTohic** [comment]
+1 tested quickly via this loader and it fixed `window is not defined` using `renderToPipeableStream` from `react-dom/server` with Express

```js
{
  test: /node_modules[\\/]@ant-design[\\/]cssinjs[\\/]es[\\/]hooks[\\/]useHMR\.js$/, // https://github.com/ant-design/cssinjs/issues/133
  loader: "string-replace-loader",
  options: {
	  search: "window",
	  replace: "globalThis",
	  flags: "g",
  },
},
```

**Comment by antoniopresto** [comment]
Hi! Do you need any help on this?

**Pull request merged**